packages/office-ui-fabric-react/src/common/DirectionalHint.ts

```diff
@@ -1,71 +1,73 @@
-export const enum DirectionalHint {
+export const DirectionalHint = {
   /**
    * Appear above the target element, with the left edges of the callout and target aligning.
    */
-  topLeftEdge = 0,
+  topLeftEdge: 0 as 0,
 
   /**
    * Appear above the target element, with the centers of the callout and target aligning.
    */
-  topCenter = 1,
+  topCenter: 1 as 1,
 
   /**
    * Appear above the target element, with the right edges of the callout and target aligning.
    */
-  topRightEdge = 2,
+  topRightEdge: 2 as 2,
 
   /**
    * Appear above the target element, aligning with the target element such that the callout tends toward the center of the screen.
    */
-  topAutoEdge = 3,
+  topAutoEdge: 3 as 3,
 
   /**
    * Appear below the target element, with the left edges of the callout and target aligning.
    */
-  bottomLeftEdge = 4,
+  bottomLeftEdge: 4 as 4,
```

Comment:
wow this is the weirdest workaround. We should show Daniel this one.

Reply:
Hmm this is not surprising to me - TypeScript widening number literals into number is in fact quite intuitive (most of the time this is the right behavior).

What we're doing here on the other hand is a little odder. 

There is a way to avoid doing this `as` on each number literal, which is to define the entire DirectionalHint as a type first and doing `const DirectionHint: DirectionHint = { ... }`. In this case, this `as` on each member will not be needed.

Reply:
Yes, and if it doesnt' work, You just need to do `const Foo: Foo = { ... } as Foo`

Reply:
@dzearing if TypeScript did not widen number literals into numbers, this mean if you write code like this:

```ts
const foo = { 
  bar: 3
}
```

You can no longer do 

```ts
foo.bar = 4
// Error: number (or 4) is not assignable to property "bar" of type "3"
```

Reply:
That does work - I've had this as my first brush with it, but figured that it's terrible devexp because every time you add a new value, you have to add it in two separate increasing long lists. I suppose I personally preferred just being able to add it one line at a time.

Reply:
I am fine with the way you did it. This is just more in response to David finding it really weird.

Reply:
For those who are curious, I love this TS deepdive book. I found my inspiration from here: https://basarat.gitbooks.io/typescript/docs/types/literal-types.html